src/runtime_src/core/edge/sw_emu/generic_pcie_hal2/shim.cxx

```diff
@@ -999,13 +1004,20 @@ namespace xclswemuhal2 {
       }
     }
     //Create thread for TCP socket connection
-    std::thread tcpSockThread = std::thread(&SwEmuShim::socketConnection, this, true);
-
+    std::thread tcpSockThread = std::thread([&]()
+                                        { this->socketConnection(true); });
     bool simDontRun = xclemulation::config::getInstance()->isDontRun();
-    if (!simDontRun) {
+    if (!simDontRun)
+    {
       if (!xclswemuhal2::isRemotePortMapped) {
         xclswemuhal2::initRemotePortMap(mFpgaDevice);
       }
+      // Lets' ensure the socket thread is started, before device process starts.
+      if (mLogStream.is_open())
```

Comment:
Where is the wait call? I just see comments. Did you run the design in a loop? before creating the PR

Reply:
It is missed, as the issue is random, failure did not occur even without wait call. The code is modified now. It will be requested for review once full testing is performed.